packages/react-storage/src/components/StorageBrowser/Views/Controls/Table.tsx

```diff
@@ -113,13 +108,20 @@ export interface TableControl<
     | 'SortAscendingIcon'
     | 'SortDescendingIcon'
   > {
-  (): React.JSX.Element;
+  <U extends LocationAccess<Permission>>(
+    props: TableControlProps<U>
+  ): React.JSX.Element;
 }
 
-export const TableControl: TableControl = () => {
-  // Data should be coming from context
-  const columns: Column<Data>[] = [];
-  const rows: Data[] = [];
+export const TableControl: TableControl = <
+  U extends LocationAccess<Permission>,
+>({
+  rows,
+  columns,
+}: TableControlProps<U>) => {
+  // @TODO data should come from context
+
+  const [, handleUpdateState] = useControl({ type: 'NAVIGATE' });
```

Comment:
Referencing handleUpdateState for just the `locations` use case complicates reusability of the `Table` for the other Views